runanalysistask/femto/RunAnalysisCharmFemtoTask.cc

```diff
@@ -124,8 +126,16 @@ void RunAnalysisCharmFemtoTask(TString configfilename, TString runMode = "full",
         TChain *chainAODfriend = new TChain("aodTree");
 
         // add a few files to the chain (change this so that your local files are added)
-        chainAOD->Add(Form("%s/AliAOD.root", pathToLocalAODfiles.data()));
-        chainAODfriend->Add(Form("%s/AliAOD.VertexingHF.root", pathToLocalAODfiles.data()));
+  
+        for (const auto &path : std::filesystem::recursive_directory_iterator(pathToLocalAODfiles.data())) {
+            if (std::string(path.path()).find("AliAOD.root") != std::string::npos) {
```

Comment:
I would create a variable with the content of `std::string(path.path())` since it is repeated many times, also `path.path()` can be replaced by this new variable